Add subscription/defer observability attributes and metrics (#8858)
apollographql/router: docs/source/routing/operations/subscriptions/configuration.mdx

  #highlight-end
```

> If a client attempts to execute a subscription on your router when it's already at `max_opened_subscriptions`, the router rejects the client's request with an error.

Author: ```suggestion
If a client attempts to execute a subscription on Apollo Router when it's already at `max_opened_subscriptions`, the router rejects the client's request with an error.
```

```

If a client attempts to execute a subscription on your router when it's already at `max_open_subscriptions`, the router rejects the client's request with an error.

### Subscription end reason telemetry

The router records an `apollo.subscription.end_reason` attribute on the `router` span for subscription operations. This attribute indicates why a subscription ended, which is useful for debugging and monitoring subscription lifecycle in your observability tools.

> The possible values for `apollo.subscription.end_reason` are:

Reviewer: ```suggestion
The `apollo.subscription.end_reason` attribute can have these values:
```

| Value | Description |
|-------|-------------|
| `server_close` | The server (subgraph) closed the connection gracefully. This occurs when the subgraph ends the subscription. |
| `stream_end` | The underlying data stream ended. This typically happens when the subgraph closes the WebSocket connection or the stream source terminates. |
| `heartbeat_delivery_failed` | A heartbeat message could not be delivered to the client. This usually indicates the client disconnected unexpectedly while the router was sending a keepalive heartbeat. |
| `client_disconnect` | The client disconnected after receiving subscription data (not during a heartbeat). This indicates the client closed the connection mid-subscription. |
| `schema_reload` | The subscription was terminated because the router's supergraph schema was updated. Clients can reconnect to resume the subscription. |
| `config_reload` | The subscription was terminated because the router's configuration was updated. Clients can reconnect to resume the subscription. |

#### Example: Monitoring subscription health

> You can use the `apollo.subscription.end_reason` attribute to monitor the health of your subscriptions:

Reviewer: ```suggestion
Use the `apollo.subscription.end_reason` attribute to monitor the health of your subscriptions:
```
